src/components/views/CourseScheduler/Course.jsx

```diff
@@ -208,19 +208,22 @@ const Course = ({
   };
 
   const semesterInNumber = () => {
+    // this is currently only stable for 2024 Winter/Spring (NOT 2023 Fall, which is part of same academic year)
+    // this controls the termID flag in the HTTP request, which Follett changed recently and may change again
+    const yearVal = 100080547 + course.year;
     switch (course.semester) {
       case "Fall":
-        return 1001 + 10 * (course.year % 100);
+        return yearVal - 2; // this does not currently work!
       case "Winter":
-        return 1002 + 10 * (course.year % 100);
+        return yearVal;
       default:
-        return 1003 + 10 * (course.year % 100);
+        return yearVal + 1;
     }
   };
 
   const bookstoreLink = () => {
     const baseBookstoreLink =
-      "https://www.bkstr.com/williamsstore/follett-discover-view/booklook?shopBy=discoverViewCourse&bookstoreId=506&divisionDisplayName=";
+      "https://www.bkstr.com/williamsstore/course-materials-results?shopBy=course&divisionDisplayName=&bookstoreId=506&programID=912";
```

Comment:
I tried experimenting with the old `follett-discover-view` link, only to realize that they have been completely deprecated.

However, the official Williams catalog is still using them...
For example, try visit [the catalog page for PHIL 225](https://catalog.williams.edu/phil/detail/?strm=1243&cn=225&crsid=020228&req_year=0) and click "View Book Information". The information will be empty.
However, if you use the new `course-materials-result` link ([link](https://www.bkstr.com/williamsstore/course-materials-results?shopBy=course&divisionDisplayName=&departmentDisplayName=PHIL&courseDisplayName=225&sectionDisplayName=T1&programId=912&termId=100082572)), you will find a list of required books.
We should perhaps also report this to whoever is managing the catalog.